Modifiers/ReportCardModifier.php

```diff
@@ -0,0 +1,144 @@
+            $line['nota4'] = ($value['nota4num'] < $examAverage ? "<b> $score4 </b>" : $score4);
+            $line['media_recuperacao'] = $examAverage;
+            $line['media'] = $average;
+            $line['curso_hora_falta'] = $absenceHours;
+            $line['frequencia'] = $this->getAttendanceByDiscipline($totalAbsencesByDiscipline, $absenceHours, $workloadByDiscipline);
+            $line['nota_exame'] = $this->args['emitir_nota_exame'] ? $this->getFormattedScore($value['nota_exame'], $value['qtd_casas_decimais']) : null;
+
+            $absenceType = $line['tipo_presenca'];
+            $schoolDays = $line['dias_letivos'];
+            $workload = $line['carga_horaria_serie'];
+            $grandTotalOfAbsences = $absenceType == RegraAvaliacao_Model_TipoPresenca::GERAL ? $value['total_faltas'] : $value['total_geral_faltas_componente'] ;
+
+            $line['media_frequencia'] = $this->getAttendance($absenceType, $grandTotalOfAbsences, $schoolDays, $workload, $absenceHours);
+            
+            $numericScores = [
+                $value['nota1num'],
+                $value['nota2num'],
+                $value['nota3num'],
+                $value['nota4num'],
+            ];
+            $line['media_grafico'] = $this->calculatesAverageForTheGraph($numericScores);
+            $line['resultado_exame'] = $this->args['termo_recuperacao_final'];
+            $line['legenda_notas'] = $scoreCaption;
+
+            $data['main'][$key] = $line;
+        }
+
+        return $data;
+    }
+
+    public function getAttendance($absenceType, $totalAbsences, $schoolDays, $workload, $absenceHours)
+    {
+        if ($absenceType == RegraAvaliacao_Model_TipoPresenca::GERAL) {
+            return bcdiv(((($schoolDays - $totalAbsences) * 100) / $schoolDays), 1, 1);
+        }
+
+        return bcdiv(100 - (($totalAbsences * ($absenceHours * 100)) / $workload), 1, 1);
+    }
+
+    public function getAttendanceByDiscipline($totalAbsencesByDiscipline, $absenceHours, $workloadByDiscipline) {
+        if (!$totalAbsencesByDiscipline) {
+            return 100.0;
+        }
+
+        if (empty($workloadByDiscipline) || $workloadByDiscipline == 0) {
+            throw new Exception('Não foi possivel calcular a frequência, pois existem disciplinas sem carga horária informada.');
+        }
+
+        return bcdiv(100 - (($totalAbsencesByDiscipline * ($absenceHours * 100)) / $workloadByDiscipline), 1, 1);
+    }
+
+    public function getFormattedScore($score, $decimalPlace)
+    {
+        if (!is_numeric($score) || empty($score)) {
+            return $score;
+        }
+
+        return str_replace('.', ',', bcdiv($score, 1, $decimalPlace));
+    }
+
+    /**
+     * Calcula média usada no gráfico
+     *
+     * @param array $notas
+     * @return float|void
+     */
+    private function calculatesAverageForTheGraph($scores = [])
+    {
+        $scores = array_filter($scores);
```

Comment:
Se a nota for `0`, vai remover ela da array, desta forma vai modificar a média final.

Reply:
As notas chegam aqui como string, no caso de ser 0 vai vir como `'0.0000'` e não terá problema.